tests/Unit/PointwiseFunctions/GeneralRelativity/Test_WeylMagnetic.cpp

```diff
@@ -0,0 +1,114 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <random>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/LeviCivitaIterator.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/VectorImpl.hpp"
+#include "Framework/CheckWithRandomValues.hpp"
+#include "Framework/SetupLocalPythonEnvironment.hpp"
+#include "Helpers/DataStructures/DataBox/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/TagsDeclarations.hpp"
+#include "PointwiseFunctions/GeneralRelativity/WeylMagnetic.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+// IWYU pragma: no_include <boost/preprocessor/arithmetic/dec.hpp>
+// IWYU pragma: no_include <boost/preprocessor/repetition/enum.hpp>
+// IWYU pragma: no_include <boost/preprocessor/tuple/reverse.hpp>
+
+namespace {
+template <typename DataType>
+void make_random_tensors(
+    const gsl::not_null<tnsr::ijj<DataType, 3>*> grad_extrinsic_curvature,
+    const gsl::not_null<tnsr::ii<DataType, 3>*> spatial_metric,
+    const DataType& used_for_size) noexcept {
+  MAKE_GENERATOR(generator);
+  std::uniform_real_distribution<> distribution(-3.0, 3.0);
+  const auto nn_generator = make_not_null(&generator);
```

Comment:
omit these extra variables -- I understand the motivation to make the call sites more succinct, but part of the motivation for the `make_not_null(&...)` arguments is that the syntax then makes it apparent at the function call which of the variables is getting mutated by the function, and this style of temporaries makes that less clear.